src/process/status.cr

```diff
@@ -16,11 +99,66 @@ class Process::Status
     end
   {% end %}
 
+  # Returns a platform-independent reason that the process terminated.
+  def exit_reason : ExitReason
+    {% if flag?(:win32) %}
+      # TODO: perhaps this should cover everything that SEH can handle?
+      # https://learn.microsoft.com/en-us/windows/win32/debug/getexceptioncode
+      case @exit_status
+      when LibC::STATUS_FATAL_APP_EXIT
+        ExitReason::Aborted
+      when LibC::STATUS_CONTROL_C_EXIT
+        ExitReason::Interrupted
+      when LibC::STATUS_BREAKPOINT
+        ExitReason::Breakpoint
+      when LibC::STATUS_ACCESS_VIOLATION, LibC::STATUS_STACK_OVERFLOW
+        ExitReason::AccessViolation
+      when LibC::STATUS_DATATYPE_MISALIGNMENT
+        ExitReason::BadMemoryAccess
+      when LibC::STATUS_ILLEGAL_INSTRUCTION, LibC::STATUS_PRIVILEGED_INSTRUCTION
+        ExitReason::BadInstruction
+      when LibC::STATUS_FLOAT_DIVIDE_BY_ZERO, LibC::STATUS_FLOAT_INEXACT_RESULT, LibC::STATUS_FLOAT_INVALID_OPERATION, LibC::STATUS_FLOAT_OVERFLOW, LibC::STATUS_FLOAT_UNDERFLOW
+        ExitReason::FloatException
+      else
+        @exit_status & 0xC0000000_u32 == 0 ? ExitReason::Normal : ExitReason::Unknown
+      end
+    {% elsif flag?(:unix) && !flag?(:wasm32) %}
+      if normal_exit?
+        ExitReason::Normal
+      elsif signal_exit?
+        case Signal.from_value?(signal_code)
+        when Nil
+          ExitReason::Signal
+        when .abrt?, .hup?, .kill?, .quit?, .term?
+          ExitReason::Aborted
+        when .int?
+          ExitReason::Interrupted
+        when .trap?
+          ExitReason::Breakpoint
+        when .segv?
+          ExitReason::AccessViolation
+        when .bus?
+          ExitReason::BadMemoryAccess
+        when .ill?
+          ExitReason::BadInstruction
+        when .fpe?
+          ExitReason::FloatException
+        else
+          ExitReason::Signal
+        end
+      else
+        # TODO: stop / continue
+        ExitReason::Unknown
+      end
+    {% else %}
+      ExitReason::Normal
```

Comment:
This makes me a bit uncomfortable. Effectively, the only platform this branch matches is `wasm32`.
I think I'd prefer to raise `NotImplementedError`. WDYT?

Reply:
@lbguilherme probably has a better idea about abnormal termination conditions there

Reply:
Given that neither WASI nor Emscripten have the concept of processes and there is no way of spawning one, I would argue for raising NotImplementedError here.